src/Model/batchprocessing/BatchProcessKaplanMeier.py

```diff
@@ -0,0 +1,163 @@
+        self.kaplanmeier_ = kaplanmeier_targetCol
+        self.kaplanmeier_targetCol = kaplanmeier_targetCol
+        self.kaplanmeier_DurationOfLifeCol = kaplanmeier_DurationOfLifeCol
+        self.kaplanmeier_AliveOrDeadCol = kaplanmeier_AliveOrDeadCol
+        self.dataDict = data_Dict
+
+    def start(self):
+        """
+        Goes through the steps of the ClinicalData-SR2CSV conversion.
+        :return: True if successful, False if not.
+        """
+        # Stop loading
+        if self.interrupt_flag.is_set():
+            self.patient_dict_container.clear()
+            self.summary = "INTERRUPT"
+            return False
+
+        if not self.ready:
+            self.summary = "SKIP"
+            return False
+
+        # See if SR contains clinical data
+        self.progress_callback.emit(("Checking SR file...", 20))
+        cd_sr = self.find_clinical_data_sr()
+
+        if cd_sr is None:
+            self.summary = "CD_NO_SR"
+            return False
+
+        # Stop loading
+        if self.interrupt_flag.is_set():
+            self.patient_dict_container.clear()
+            self.summary = "INTERRUPT"
+            return False
+
+        # Read in clinical data from SR
+        self.progress_callback.emit(("Reading clinical data...", 50))
+        data_dict = self.read_clinical_data_from_sr(cd_sr)
+
+        # Stop loading
+        if self.interrupt_flag.is_set():
+            self.patient_dict_container.clear()
+            self.summary = "INTERRUPT"
+            return False
+
+        # Write clinical data to CSV
+        self.progress_callback.emit(("Writing clinical data to CSV...", 80))
+        self.plot(data_dict, self.kaplanmeier_targetCol, self.kaplanmeier_DurationOfLifeCol, self.kaplanmeier_AliveOrDeadCol)
+        return True
+
+    def find_clinical_data_sr(self):
+        """
+        Searches the patient dict container for any SR files containing
+        clinical data. Returns the first SR with clinical data found.
+        :return: ds, SR dataset containing clinical data, or None if
+                 nothing found.
+        """
+        datasets = self.patient_dict_container.dataset
+
+        for ds in datasets:
+            # Check for SR files
+            if datasets[ds].SOPClassUID == '1.2.840.10008.5.1.4.1.1.88.33':
+                # Check to see if it is a clinical data SR
+                if datasets[ds].SeriesDescription == "CLINICAL-DATA":
+                    return datasets[ds]
+
+        return None
+
+    def read_clinical_data_from_sr(self, sr_cd):
+        """
+        Reads clinical data from the found SR file.
+        :param sr_cd: the clinical data SR dataset.
+        :return: dictionary of clinical data, where keys are attributes
+                 and values are data.
+        """
+        data = sr_cd.ContentSequence[0].TextValue
+
+        data_dict = {}
+
+        data_list = data.split("\n")
+        for row in range(len(data_list)):
```

Comment:
This should probably just be iteration over the `data_list` instead of using range. `for row in data_list:` or something similar